oasislmf/pytools/gul/core.py

```diff
@@ -116,6 +116,33 @@ def split_tiv(gulitems, tiv):
             gulitems[j] *= f
 
 
+@njit(cache=True, fastmath=True)
+def split_tiv_multiplicative(gulitems, tiv):
+    """Split the total insured value (TIV) using a multiplicative formula for the
+    total loss as tiv * (1 - (1-A)*(1-B)*(1-C)...), where A, B, C are damage ratios
+    computed as the ratio between a sub-peril loss and the tiv. Sub-peril losses
+    in gulitems are always back-allocated proportionally to the losses.
+
+    Args:
+        gulitems (numpy.array[oasis_float]): array containing losses of all items.
+        tiv (oasis_float): total insured value.
+    """
+    Ngulitems = gulitems.shape[0]
+    total_loss = 1.
+    for i in range(Ngulitems):
+        total_loss *= 1. - gulitems[i] / tiv
+
+    total_loss = 1. - total_loss
+    total_loss *= tiv
+    if total_loss > 0.:
```

Comment:
I think this is the same and a bit less computation:
if insure_value_left < 1:
          f = 1/(1-insure_value_left)
          for j in range(Ngulitems):
            # editing in-place the np array
            gulitems[j] *= f
Can you double check?

Reply:
doesn't apply anymore I think